gen3sdk/R/Gen3File.R

```diff
@@ -0,0 +1,57 @@
+library(httr)
+library(jsonlite)
+
+Gen3File <- setRefClass("Gen3File",
+# For interacting with Gen3 file management features.
+
+# A class for interacting with the Gen3 file download services.
+# Supports getting presigned urls right now.
+
+# Args:
+#     endpoint (str): The URL of the data commons.
+#     auth_provider (Gen3Auth): A Gen3Auth class instance.
+
+# Examples:
+#     This generates the Gen3File class pointed at the sandbox commons while
+#     using the credentials.json downloaded from the commons profile page.
+
+#     >>> endpoint <- "https://nci-crdc-demo.datacommons.io"
+#     ... auth <- Gen3AuthHelper(endpoint, refresh_file="credentials.json")
+#     ... file <- Gen3File(endpoint, auth)
+
+    fields = list(
+        endpoint= "character",
+        auth_provider= "Gen3AuthHelper"
+    ),
+
+    methods = list(
+        initialize = function(endpoint, auth_provider) {
+            .self$endpoint <- endpoint
```

Comment:
again here should check if these look reasonable

Reply:
Initialize method automatically throws an error when parameter is missing
Error in .Object$initialize(...) : argument "endpoint" is missing, with no default